docling/backend/xml_backend.py

```diff
@@ -0,0 +1,296 @@
+            for affiliation in affiliations_:
+                authors_affiliations.append(affiliation["name"])
+
+        doc.add_text(
+            parent=None,
+            text="; ".join(authors_affiliations),
+            label=DocItemLabel.PARAGRAPH,
+        )
+        return
+
+    def add_abstract(self, doc: DoclingDocument, xml_components: dict) -> None:
+        abstract_text: str = (
+            xml_components["info"]["abstract"].replace("\n", " ").strip()
+        )
+        if abstract_text.strip():
+            doc.add_text(text=abstract_text, label=DocItemLabel.TEXT)
+        return
+
+    def add_main_text(self, doc: DoclingDocument, xml_components: dict) -> None:
+        sections: list = []
+        parent = None
+        for paragraph in xml_components["paragraphs"]:
+            if ("section" in paragraph) and (paragraph["section"] == ""):
+                continue
+
+            if "section" in paragraph and paragraph["section"] not in sections:
+                section: str = paragraph["section"].replace("\n", " ").strip()
+                sections.append(section)
+                if section in self.parents:
+                    parent = self.parents[section]
+                else:
+                    parent = None
+
+                self.parents[section] = doc.add_heading(parent=parent, text=section)
+
+            if "text" in paragraph:
+                text: str = paragraph["text"].replace("\n", " ").strip()
+
+                if paragraph["section"] in self.parents:
+                    parent = self.parents[paragraph["section"]]
+                else:
+                    parent = None
+
+                doc.add_text(parent=parent, label=DocItemLabel.TEXT, text=text)
+        return
+
+    def add_references(self, doc: DoclingDocument, xml_components: dict) -> None:
+        doc.add_heading(parent=None, text="References")
+        current_list = doc.add_group(label=GroupLabel.LIST, name="list")
+        for reference in xml_components["references"]:
+            reference_text: str = (
+                reference["name"]
+                + ". "
+                + reference["article_title"]
+                + ". "
+                + reference["journal"]
+                + " ("
+                + reference["year"]
+                + ")"
+            )
+            doc.add_list_item(
+                text=reference_text, enumerated=False, parent=current_list
+            )
+        return
+
+    def add_tables(self, doc: DoclingDocument, xml_components: dict) -> None:
+        doc.add_heading(parent=None, text="Tables")
+        for table_xml_component in xml_components["tables"]:
+            try:
+                self.add_table(doc, table_xml_component)
+            except Exception as e:
+                print(f"Skipping unsupported table for: {str(self.file)}")
+                pass
+        return
+
+    def add_table(self, doc: DoclingDocument, table_xml_component: dict) -> None:
```

Comment:
Can we reuse the code from the HTML backend implementation instead of duplicating it?